src/libstd/sys_common/backtrace.rs

```diff
@@ -104,56 +104,52 @@ fn filter_frames(frames: &[Frame],
     // We want to filter out frames with some prefixes
     // from both top and bottom of the call stack.
     static BAD_PREFIXES_TOP: &'static [&'static str] = &[
-        "_ZN3std3sys3imp9backtrace",
+        // std::sys::imp::backtrace
         "ZN3std3sys3imp9backtrace",
-        "std::sys::imp::backtrace",
-        "_ZN3std10sys_common9backtrace",
+        // std::sys_common::backtrace
         "ZN3std10sys_common9backtrace",
-        "std::sys_common::backtrace",
-        "_ZN3std9panicking",
+        // std::panicking
         "ZN3std9panicking",
-        "std::panicking",
-        "_ZN4core9panicking",
+        // core::panicking
         "ZN4core9panicking",
-        "core::panicking",
-        "_ZN4core6result13unwrap_failed",
+        // core::result::unwrap_failed
         "ZN4core6result13unwrap_failed",
-        "core::result::unwrap_failed",
+
         "rust_begin_unwind",
         "_ZN4drop",
         "mingw_set_invalid_parameter_handler",
     ];
     static BAD_PREFIXES_BOTTOM: &'static [&'static str] = &[
-        "_ZN3std9panicking",
+        // core::panicking
+        "ZN4core9panicking",
+        // std::panicking
         "ZN3std9panicking",
-        "std::panicking",
-        "_ZN3std5panic",
+        // std::panic
         "ZN3std5panic",
-        "std::panic",
-        "_ZN4core9panicking",
-        "ZN4core9panicking",
-        "core::panicking",
-        "_ZN3std2rt10lang_start",
-        "ZN3std2rt10lang_start",
-        "std::rt::lang_start",
-        "panic_unwind::__rust_maybe_catch_panic",
-        "__rust_maybe_catch_panic",
-        "_rust_maybe_catch_panic",
-        "__libc_start_main",
+        // core::ops::FnOnce::call_once
+        "ZN4core3ops6FnOnce9call_once",
+
         "__rust_try",
-        "_start",
-        "main",
         "BaseThreadInitThunk",
         "RtlInitializeExceptionChain",
         "__scrt_common_main_seh",
         "_ZN4drop",
```

Comment:
IIRC, `_ZN4drop` were some kind of corrupted misinterpreted frames on 32-bit mingw (similar to `mingw_set_invalid_parameter_handler`). Not sure how reproducible they are on executables other than `run-pass/bactrace.rs` test.